listenbrainz_spark/ftp/download.py

```diff
@@ -0,0 +1,133 @@
+        """
+        if dump_id:
+            for dump_name in dump:
+                if int(dump_name.split('-')[dump_id_pos] == dump_id):
+                    req_dump = dump_name
+                    break
+                else:
+                    err_msg = "Could not find dump with ID: {}. Aborting...".format(dump_id)
+                    raise DumpNotFoundException(err_msg)
+        else:
+            req_dump = dump[-1]
+        return req_dump
+
+    def get_file_name(self, dump_name):
+        """ Get the name of the Spark dump archive from the dump directory name.
+
+            Args:
+                dump_name (str): FTP dump directory name.
+
+            Returns:
+                '' : Spark dump archive name.
+        """
+        return dump_name + '.tar.bz2'
+
+    def get_listens_file_name(self):
+        """ Get the name of Spark listens dump name archive.
+
+            Returns:
+                '' : Spark listens dump archive name.
+        """
+        return config.TEMP_LISTENS_DUMP
+
+    def download_msid_mbid_mapping(self, directory, mapping_dump_id=None):
+        """ Download msid_mbid_mapping to dir passed as an argument.
+
+            Args:
+                directory (str): Dir to save mappings locally.
+                mapping_dump_id (int): Unique identifier of mapping to be downloaded.
+                    If not provided, most recent mapping will be downloaded.
+
+            Returns:
+                dest_path (str): Local path where mapping has been downloaded.
+        """
+        self.connection.cwd('/pub/musicbrainz/listenbrainz/labs/mappings/msid-mbid-mapping/')
+        mappings_dump = self.list_dir()
+        req_mapping_dump = self.get_req_dump(mappings_dump, mapping_dump_id, 3)
+
+        self.connection.cwd(req_mapping_dump)
+        mapping_file_name = self.get_file_name(req_mapping_dump)
+
+        t0 = time()
+        current_app.logger.info('Downloading {} from FTP...'.format(mapping_file_name))
+        dest_path = self.download_dump(mapping_file_name, directory)
+        current_app.logger.info('Done. Total time: {:.2f} sec'.format(time() - t0))
+        return dest_path
+
+    def download_msid_mbid_mapping_with_text(self):
+        pass
+
+    def download_msid_mbid_mapping_with_matchable(self):
+        pass
+
+    def download_listens(self, directory, listens_dump_id=None):
+        """ Download listens to dir passed as an argument.
+
+            Args:
+                directory (str): Dir to save listens locally.
+                listens_dump_id (int): Unique identifier of listens to be downloaded.
+                    If not provided, most recent listens will be downloaded.
+
+            Returns:
+                dest_path (str): Local path where listens have been downloaded.
+        """
+        self.connection.cwd('/pub/musicbrainz/listenbrainz/fullexport/')
+        listens_dump = self.list_dir()
+        # Temporary listens dump name.
```

Comment:
Is this temporary since the dumps are currently borked?

Reply:
Yes. We are not fetching the recent dump name like in mappings and artist relation because the recent full dump is empty.
http://ftp.musicbrainz.org/pub/musicbrainz/listenbrainz/fullexport/listenbrainz-dump-97-20191031-000002-full/

Till we are clear about the full/incremental dumps I have kept temp name so that new contributors have some data to play with